docs/function-buffer-varints.md

```diff
@@ -0,0 +1,70 @@
+# Varint Functions for Buffers
+
+## Summary
+
+This RFC proposes the addition of new functions to the buffer API:
+
+- `buffer.readleb128` 
+- `buffer.readleb128`
+- `buffer.readuleb128`
+- `buffer.writeuleb128` 
+
+These functions will write and read (U)LEB-128 variable-width integers (varints) from a buffer.
+
+## Motivation
+
+One of the top use cases for buffers is compressing and serializing data as small as possible. Numbers consume 8 bytes in their raw form but can be serialized in smaller sizes using buffers to reduce size. However, writing code to serialize as the smallest size can be tedious and confusing for less-experienced programmers.  
+
+The following are implementations of ULEB-128 reading/writing in Luau:
+
+```lua
+local function writeuleb128(stream, value)
+  while value >= 0x80 do
+    buffer.writeu8(bit32.bor(bit32.band(value, 0x7f), 0x80))
+    value = bit32.rshift(value, 7)
+  end
+  buffer.writeu8(value)
+end
+```
+
+```lua
+local function readuleb128(stream)
+  local result = 0
+  local bits = 0
+  while true do
+    local byte = buffer.readu8(stream)
+    result = bit32.bor(result, bit32.lshift(bit32.band(byte, 0x7f), bits))
+    if byte < 0x80 then
+      break
+    end
+    bits = bits + 7
+  end
+  return result
+end
+```
+
+The functions above are inefficient and difficult to understand compared to a native implementation. Implementations will also be needed for the corresponding signed functions.
```

Comment:
Please use 'library implementation' instead of 'built-in' (implies fastcalls) and 'native' (implies codegen).

And if fastcall/native was implied, we are unlikely to accept this.
Reason for that is that almost everything can be faster with a built-in or with native codegen, so it cannot be the only motivation to having it implemented that way.
It has to be supported by real-life use cases that are suffering by performance that regular C library implementation is able to provide or some other good reason.